src/core/crypto/ecdsa.cpp

```diff
@@ -117,18 +175,47 @@ Error P256::KeyPair::GetPublicKey(PublicKey &aPublicKey) const
 exit:
     mbedtls_pk_free(&pk);
     return error;
+
+#endif
 }
 
 Error P256::KeyPair::Sign(const Sha256::Hash &aHash, Signature &aSignature) const
 {
+#if defined(MBEDTLS_USE_TINYCRYPT)
+
     Error                 error;
     mbedtls_pk_context    pk;
-    mbedtls_ecp_keypair * keypair;
-    mbedtls_ecdsa_context ecdsa;
-    mbedtls_mpi           r;
-    mbedtls_mpi           s;
+    mbedtls_uecc_keypair *keypair;
     int                   ret;
 
+    uint8_t *sig = (uint8_t *)malloc(2 * kMpiSize);
```

Comment:
We cannot and should not use `malloc()` and `free()` lib functions in OT (to avoid depedency to libraries and keep OT portable on many platforms). We have `heap.h` (and `heap.hpp` definitions) which provide platform specific versions.

But actually in the case here, I don't think we need heap allocated buffer and can use a stack variable (array):
```c++
uint8_t sig[2*kMpiSize];
```